kernel/threads.c

```diff
@@ -21,6 +21,12 @@ static vmcb vmcbs[NUMVMSPACES];
 static tcb tcbs[NUMTHREADS];
 static tcb *idle_tcb = 0;
 
+void save_thread_registers (struct all_registers* all_registers) {
```

Comment:
nit: I feel sort of like if you call save_thread_registers, you should later clear them so that all_registers is null instead of pointing to random garbage, if you later call something where you didn't bother to save the registers for whatever reason.

Reply:
As above with my 'fork' comment, this gets better 3 CLs from now, when 'struct all_registers' is just copied into the TCB, so I don't think it's worth tracking down all the times that pointer should be reset.